fzy-lua fails to install on Windows

Hey :wave: 

Attepting `luarocks install fzy` on Windows fails with

```
src/match.c(18): fatal error C1083: Cannot open include file: 'strings.h': No such file or directory
```

https://github.com/nvim-neorocks/rocks-binaries/actions/runs/8191805472/job/22401866175#step:9:815

There are two ways I could think of to fix this:

1. Use a [platform override](https://github.com/luarocks/luarocks/wiki/platform-overrides) in the rockspec to prevent it from trying to compile the native library.
2. Add an adapter header ([see this PR](https://github.com/google/gumbo-parser/pull/429/files)). I'm not sure if there are any other headers that might be missing on Windows though.

--- a/.github/workflows/luarocks.yml
+++ b/.github/workflows/luarocks.yml
@@ -1,0 +1,28 @@
+---
+name: luarocks upload
+on:
+  push:
+    tags:
+      - 'v*'
+  pull_request: # Will test packaging without publishing on PR
+  workflow_dispatch: # Allow manual trigger
+
+jobs:
+  luarocks-release:
+    runs-on: ubuntu-latest
+    name: luarocks upload
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Luarocks Upload
+        uses: nvim-neorocks/luarocks-tag-release@v5
+        env:
+          LUAROCKS_API_KEY: ${{ secrets.LUAROCKS_API_KEY }}
+        with:
+          summary: "A lua implementation of the fzy fuzzy matching algorithm"
+          detailed_description: |
+            A Lua port of fzy's fuzzy string matching algorithm.
+            This includes both a pure Lua implementation and a compiled C implementation with a Lua wrapper.
+          copy_directories: |
+            docs
+          test_interpreters: "lua"

